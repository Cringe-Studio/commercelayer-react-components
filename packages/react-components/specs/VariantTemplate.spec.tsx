import React from 'react'
import VariantTemplate, {
  propTypes,
} from '../#components/utils/VariantTemplate'
import renderer from 'react-test-renderer'

test('<VariantTemplate/>', () => {
  expect.assertions(9)
  const component = renderer.create(
    <VariantTemplate
      options={[
        {
          label: '6 months',
          code: 'BABYONBU000000E63E746MXX',
        },
      ]}
      variants={{}}
    />
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(propTypes.children)
  expect(proptypes.placeholder).toBe(propTypes.placeholder)
  expect(proptypes.options).toBe(propTypes.options)
  expect(proptypes.name).toBe(propTypes.name)
  expect(proptypes.type).toBe(propTypes.type)
  expect(proptypes.skuCode).toBe(propTypes.skuCode)
  expect(proptypes.variants).toBe(propTypes.variants)
  expect(proptypes.onChange).toBe(propTypes.onChange)
})
