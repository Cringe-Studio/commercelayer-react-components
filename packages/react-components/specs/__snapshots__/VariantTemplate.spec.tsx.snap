// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VariantTemplate/> 1`] = `null`;
